lib/teleterm/daemon/daemon.go

```diff
@@ -148,6 +152,35 @@ func (s *Service) createGateway(ctx context.Context, params CreateGatewayParams)
 		LocalPort:             params.LocalPort,
 		CLICommandProvider:    cliCommandProvider,
 		TCPPortAllocator:      s.cfg.TCPPortAllocator,
+		// TODO: Add test that makes sure NewWithLocalPort properly copies OnNewConnection.
+		OnNewConnection: func(gatewayURI uri.ResourceURI, targetURI string) {
+			// TODO Refactor targetURI from string to uri.ResourceURI.
+			// https://github.com/gravitational/teleport/issues/15953
+			clusterURI, err := uri.ParseClusterURI(targetURI)
+			if err != nil {
+				s.Log().WithError(err).Warnf(
+					"Malformed gateway target URI (%s), could not parse it into uri.ResourceURI",
+					targetURI,
+				)
+				return
+			}
+
+			clusterEvent := &api.ClusterEvent{
+				ClusterUri: clusterURI.String(),
+				Event: &api.ClusterEvent_NewGatewayConnectionAccepted{
+					NewGatewayConnectionAccepted: &api.NewGatewayConnectionAccepted{
+						GatewayUri: gatewayURI.String(),
+						TargetUri:  targetURI,
+					},
+				},
+			}
+
+			s.Log().Debug("Sending ClusterEvent to clusterEventsC")
+
+			s.clusterEventsC <- clusterEvent
+
+			s.Log().Debug("Message sent to clusterEventsC")
```

Comment:
I might as well remove those debugs in the final version. I kept them here to debug some deadlocks that I encountered while writing the first version of this code.